modules/serverless-kafka-source-kn.adoc

```diff
@@ -0,0 +1,100 @@
+//
+// * serverless/event_sources/serverless-kafka-source.adoc
+[id="serverless-kafka-source-kn_{context}"]
+= Creating a Kafka event source using the `kn` CLI
+
+This section describes how to create a KafkaSource using the `kn` command.
+
+.Prerequisites
+
+* The {ServerlessOperatorName}, `KnativeEventing`, and the `KnativeKafka` custom resource definitions are installed on your cluster.
+* You have created a project or have access to a project with the appropriate roles and permissions to create applications and other workloads in {product-title}.
+* You have access to the Kafka cluster that produces the Kafka messages you want to import.
+
+.Procedure
+
+. To verify that the Kafka source is working, create a Knative service that dumps incoming messages into the service's logs:
++
+[source, terminal]
+----
+$ kn service create event-display \
+    --image quay.io/openshift-knative/knative-eventing-sources-event-display:latest
+----
+
+. Create a KafkaSource:
++
+[source,terminal]
+----
+$ kn source kafka create mykafkasrc \
+    --servers my-cluster-kafka-bootstrap.kafka.svc:9092 \
+    --topics test-topic --consumergroup test-consumer-group \
+    --sink event-display
+----
++
+The `--servers`, `--topics`, and `--consumergroup` options specify the connection parameters to the Kafka cluster. The `--consumergroup` option is optional.
++
+Additionally, this command makes several assumptions, including:
++
+* `namespace` is `kafka`
+* `pod` is `my-cluster-kafka-0`
+
+. Optionally, see the details about the created Kafka source:
++
+[source, terminal]
+----
+$ kn source kafka describe mykafkasrc
+Name:              mykafkasrc
+Namespace:         default
+Age:               1h
+BootstrapServers:  my-cluster-kafka-bootstrap.kafka.svc:9092
+Topics:            test-topic
+ConsumerGroup:     test-consumer-group
+
+Sink:
+  Name:       event-display
+  Namespace:  default
+  Resource:   Service (serving.knative.dev/v1)
+
+Conditions:
+  OK TYPE            AGE REASON
+  ++ Ready            1h
+  ++ Deployed         1h
+  ++ SinkProvided     1h
+----
+
+.Verification steps
+
+. Trigger the Kafka instance to send a message to the topic:
++
+[source,terminal]
+----
+$ oc -n kafka exec my-cluster-kafka-0 \
+    -ti -- bin/kafka-console-producer.sh \
+    --bootstrap-server localhost:9092 \
+    --topic test-topic
+----
++
+Enter the message in the prompt.
++
+This command makes several assumptions, including:
++
+* `namespace` is `kafka`
+* `pod` is `my-cluster-kafka-0`
++
+Alternatively, you can send a message without Kafka running in the cluster:
++
+[source,terminal]
+----
+$ oc -n kafka run kafka-producer \
+    -ti --image=quay.io/strimzi/latest-kafka-2.7.0 --rm=true \
+    --restart=Never -- bin/kafka-console-producer.sh \
+    --broker-list my-cluster-kafka-bootstrap:9092 --topic my-topic
+----
+
+. Verify that the message arrived:
++
+[source,terminal]
+----
+$ oc logs $(oc get pod -o name | grep event-display) -c user-container
+----
```

Comment:
I'd add an example output after this

Reply:
@rhuss @dsimansk Could you please provide an example output here?

Reply:
sure, let me do a review tomorrow, too. @dsimansk if you are faster to provide the example, go ahead ;-)

Reply:
```bash
➜  ~ oc logs $(oc get pod -o name | grep event-display) -c user-container
☁️  cloudevents.Event
Validation: valid
Context Attributes,
  specversion: 1.0
  type: dev.knative.kafka.event
  source: /apis/v1/namespaces/default/kafkasources/mykafkasrc#test-topic
  subject: partition:46#0
  id: partition:46/offset:0
  time: 2021-03-10T11:21:49.4Z
Extensions,
  traceparent: 00-161ff3815727d8755848ec01c866d1cd-7ff3916c44334678-00
Data,
  Hello!

```

Reply:
Thanks David!